examples/units_and_coordinates/STEREOSECCHI_starfield.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+"""
+===========================================================
+Identifying stars in a STEREO/SECCHI COR2 coronagraph image
+===========================================================
+
+How to use the Vizier star catalog to identify bright points in a
+STEREO/SECCHI COR 2 image as stars. As a bonus, we also identify Mars.
+"""
+import matplotlib.pyplot as plt
+
+from astroquery.vizier import Vizier
+import astropy.units as u
+from astropy.coordinates import SkyCoord, get_body_barycentric
+
+import sunpy.map
+from sunpy.coordinates import get_body_heliographic_stonyhurst, frames
+from sunpy.net import helioviewer
+hv = helioviewer.HelioviewerClient()
+
+###############################################################################
+# Let's download a STEREO-A SECCHI COR2 image from Helioviewer which provided
+# pre-processed images and load it into a Map.
+f = hv.download_jp2('2014/05/15 07:54', observatory='STEREO_A', instrument='SECCHI', detector='COR2')
+cor2 = sunpy.map.Map(f)
+
+###############################################################################
+# To efficiently search the star field we need to know what stars are near the
+# Sun as observed by STEREO. This can be found by find the difference vector
+# between the Sun and the location of STEREO.
+stereo = cor2.observer_coordinate.transform_to('icrs').cartesian
+sun = get_body_barycentric('sun', time=cor2.date)
+diff = sun - stereo
+search_coord = SkyCoord(diff, frame='icrs')
```

Comment:
For this particular situation, it is more straightforward to transform the STEREO coordinate to HCRS, which is precisely the Sun-STEREO vector in ICRF, and then negating that vector (to get the STEREO-Sun vector) for the search.

Reply:
The value of search coord here is
```
<SkyCoord (ICRS): (ra, dec, distance) in (deg, deg, AU)
    (210.14232004, -12.28522192, 0.96125506)>
``` 
if I do what you suggest I get
```
cor2.observer_coordinate.transform_to('hcrs')
<SkyCoord (HCRS: obstime=2014-05-15T07:54:00.005): (ra, dec, distance) in (deg, deg, m)
    (30.14232004, 12.28522192, 1.4380171e+11)>
```
which does not make sense to me.

Reply:
As I said, you also need to negate that vector, which in spherical coordinates is equivalent to adding 180 degrees to RA and negating the declination.

Reply:
This wasn't addressed, do we want it to be?